sdk/include/opentelemetry/sdk/metrics/aggregator/counter_aggregator.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include "opentelemetry/version.h"
+#include "opentelemetry/metrics/instrument.h"
+#include "opentelemetry/sdk/metrics/aggregator/aggregator.h"
+
+#include <variant>
+#include <vector>
+#include <mutex>
+#include <memory>
+
+namespace metrics_api = opentelemetry::metrics;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+
+template<class T>
+class CounterAggregator final : public Aggregator<T>
+{
+    
+public:
+    CounterAggregator(metrics_api::BoundInstrumentKind kind)
+    {
+        this->kind_ = kind;
+        this->values_     = std::vector<T>(1, 0);
+        this->checkpoint_ = std::vector<T>(1, 0);
+    }
+    
+    /**
+     * Recieves a captured value from the instrument and applies it to the current aggregator value.
+     *
+     * @param val, the raw value used in aggregation
+     * @return none
+     */
+    void update(T val) override
+    {
+        this->mu_.lock();
+        this->values_[0] += val;  // atomic operation
```

Comment:
For int and long, is it possible to spin + CAS (compare and swap) to achieve better perf?
Not blocking this PR though.